pandas/indexes/base.py

```diff
@@ -1745,29 +1745,13 @@ def argsort(self, *args, **kwargs):
         return result.argsort(*args, **kwargs)
 
     def __add__(self, other):
-        if is_list_like(other):
-            warnings.warn("using '+' to provide set union with Indexes is "
-                          "deprecated, use '|' or .union()", FutureWarning,
-                          stacklevel=2)
-        if isinstance(other, Index):
```

Comment:
Why the `isinstance` part? (we don't only allow to add an Index, but also a scalar or list/array)